controller/volume_controller.go

```diff
@@ -311,6 +315,13 @@ func (vc *VolumeController) syncVolume(key string) (err error) {
 			}
 		}
 
+		for _, snap := range snapshots {
+			if snap.DeletionTimestamp == nil {
+				if err := vc.ds.DeleteSnapshot(snap.Name); err != nil {
+					return err
+				}
+			}
+		}
```

Comment:
I was thinking about this case if we don't delete snapshot here:
1. User deletes a volume -> volume CR is deleted
2. Snapshots haven't been deleted yet
3. User create a volume with the same name
4. Snapshots might be referring to the new volume

But now rethinking about it, the case is not possible because the owerreference of the snapshot referring to the UUID of the volume. 

I will modify the logic to use `DeletePropagationForeground` thanks

Reply:
Sorry. User might use kubectl directly and Kubernetes uses background cascading deletion by default. 

Should we keep the logic here

Reply:
I mean user might delete the volume CR using kubectl and Kubernetes uses background cascading deletion by default